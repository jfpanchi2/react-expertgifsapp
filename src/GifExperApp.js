import React from 'react'
import {useState} from 'react';
import { AddCategoria } from './components/AddCategoria';
import { GifGrid } from './components/GifGrid';

export const GifExperApp = ({defaultCategories=[]}) => {
    //const categorias =['One Puch','Samurai x','DragonBall']
    //const [categorias, setCategorias] = useState(['Cats']);
    const [categorias, setCategorias] = useState(defaultCategories);
    /* const handleAdd = () =>{
        setCategorias([...categorias,'HunterxHunter']);
    } */
    return (
        <div>
            <h1>GIFS</h1>
            <AddCategoria setCategorias={setCategorias}></AddCategoria>
            <hr></hr>
            {/* <button >Agregar</button> */}
            
                {
                    categorias.map( categoria =>{
                        return <GifGrid categoria={categoria} key={categoria}></GifGrid>
                    })
                }
            
           
        </div>
    )
}
