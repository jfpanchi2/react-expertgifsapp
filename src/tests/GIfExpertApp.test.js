import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import {GifExperApp} from "../GifExperApp";

describe('Pruebas en GifExpertApp', () => {
    test('Debe mostrar el componente', () => {
        const wrapper = shallow(<GifExperApp></GifExperApp>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One punch', 'Cats'];
        const wrapper = shallow(<GifExperApp defaultCategories={categories}></GifExperApp>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
