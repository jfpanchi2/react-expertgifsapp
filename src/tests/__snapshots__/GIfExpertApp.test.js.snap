// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<div>
  <h1>
    GIFS
  </h1>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    categoria="One punch"
    key="One punch"
  />
  <GifGrid
    categoria="Cats"
    key="Cats"
  />
</div>
`;

exports[`Pruebas en GifExpertApp Debe mostrar el componente 1`] = `
<div>
  <h1>
    GIFS
  </h1>
  <AddCategoria
    setCategorias={[Function]}
  />
  <hr />
</div>
`;
